src/main/scala/org/scalajs/tools/tsimporter/sc/Definitions.scala

```diff
@@ -299,6 +299,9 @@ object TypeRef {
   val Unit = TypeRef(scala dot Name("Unit"))
   val Null = TypeRef(scala dot Name("Null"))
   val Nothing = TypeRef(scala dot Name("Nothing"))
+  val ThisType = TypeRef(new Name("this.type") {
+    override def toString(): String = "this.type"
+  })
```

Comment:
That's nonsensical. `this.type` is not the selection of a member `type` on the type `this`.

It would make more sense to represent `this.type` as a `Singleton(QualifiedName(Name.THIS))`, where `THIS` is something like `Name("<this>")`. It is then the responsibility of the printer to handle this case.

Reply:
@sjrd Done!